snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/version/SnomedVersioningRequest.java

```diff
@@ -112,14 +112,15 @@ protected void doVersionComponents(TransactionContext context) throws Exception
 		for (String module : ImmutableSet.copyOf(componentIdsByReferringModule.keySet())) {
 			final Collection<String> dependencies = componentIdsByReferringModule.removeAll(module);
 			for (Class<? extends SnomedComponentDocument> type : CORE_COMPONENT_TYPES) {
-				Query<String[]> dependencyQuery = Query.select(String[].class)
-						.from(type)
-						.fields(SnomedComponentDocument.Fields.ID, SnomedComponentDocument.Fields.MODULE_ID, SnomedComponentDocument.Fields.EFFECTIVE_TIME)
-						.where(SnomedComponentDocument.Expressions.ids(dependencies))
-						.limit(10000)
-						.build();
-				for (Hits<String[]> dependencyHits : context.service(RevisionSearcher.class).scroll(dependencyQuery)) {
-					for (String[] dependency : dependencyHits) {
+				Query.select(String[].class)
+					.from(type)
+					.fields(SnomedComponentDocument.Fields.ID, SnomedComponentDocument.Fields.MODULE_ID, SnomedComponentDocument.Fields.EFFECTIVE_TIME)
+					.where(SnomedComponentDocument.Expressions.ids(dependencies))
+					.limit(10000)
```

Comment:
It would be great to use either the low commit threshold or some other configuration key that specifies the amount of default page size when loading documents from the index. That way we could easily scale the termserver based on the current machine's hardware. Keep it like this for now, but please make a note about this somewhere.